server/segments_endpoint_test.go

```diff
@@ -0,0 +1,69 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
```

Comment:
You could test what happens when two requests with the same name are sent, one after another. That sounds like an interesting test to me.